addOns/openapi/src/main/java/org/zaproxy/zap/extension/openapi/ProgressPanel.java

```diff
@@ -0,0 +1,202 @@
+            statusCombo.addItem(Constant.messages.getString("openapi.progress.panel.status.all"));
+            statusCombo.addItem(
+                    Constant.messages.getString("openapi.progress.panel.status.completed"));
+            statusCombo.addItem(
+                    Constant.messages.getString("openapi.progress.panel.status.inprogress"));
+            statusCombo.setSelectedIndex(0);
+            // TODO : Change to not use setMaximumSize
+            statusCombo.setMaximumSize(new Dimension(150, Integer.MAX_VALUE));
+            statusCombo.addActionListener(
+                    e -> {
+                        JComboBox<String> combo = (JComboBox<String>) e.getSource();
+                        updateImportPanesPanel((String) combo.getSelectedItem());
+                    });
+
+            JLabel showLabel =
+                    new JLabel(Constant.messages.getString("openapi.progress.panel.status.show"));
+            showLabel.setLabelFor(statusCombo);
+
+            toolBar.add(clearButton);
+            toolBar.add(new Separator());
+            toolBar.add(showLabel);
+            toolBar.add(statusCombo);
+        }
+        return toolBar;
+    }
+
+    private JScrollPane getJScrollPane() {
+        if (jScrollPane == null) {
+            jScrollPane = new JScrollPane(getImportPanesPanel());
+            jScrollPane.setFont(FontUtils.getFont("Dialog"));
+            jScrollPane.setVerticalScrollBarPolicy(
+                    javax.swing.ScrollPaneConstants.VERTICAL_SCROLLBAR_AS_NEEDED);
+        }
+        return jScrollPane;
+    }
+
+    private JPanel getImportPanesPanel() {
+        if (importPanesPanel == null) {
+            importPanesPanel = new JPanel();
+            importPanesPanel.setLayout(new GridBagLayout());
+        }
+        return importPanesPanel;
+    }
+
+    private void updateImportPanesPanel(String status) {
+        getImportPanesPanel().removeAll();
+        panelConstraints.gridy = 0;
+        for (ImportPane pane : importPanes) {
+            if ((status.equals("Completed") && !pane.getProgressStatus())
+                    || (status.equals("In Progress") && pane.getProgressStatus())
+                    || status.equals("All")) {
+                getImportPanesPanel().add(pane, panelConstraints);
+                panelConstraints.gridy++;
+            }
+        }
+        getImportPanesPanel().revalidate();
+        getImportPanesPanel().repaint();
+    }
+
+    public void addImportPane(ImportPane pane) {
+        importPanes.add(pane);
+        getImportPanesPanel().add(pane, panelConstraints);
+        panelConstraints.gridy++;
+    }
+
+    private void clear() {
+        ArrayList<ImportPane> panesToRemove = new ArrayList<>();
+        for (ImportPane pane : importPanes) {
+            if (!pane.getProgressStatus()
+                    || pane.getImportStatus().getText().equals(Constant.messages.getString(
+                            "openapi.progress.importpane.status", 0, 0))) {
```

Comment:
This doesn't seem like a wonderful way to handle it, but there doesn't seem to be any other way to handle the condition. Unless we just clear everything because the user said to? (Even if one is in progress) 🤷

Reply:
Latest push should address this.

Reply:
Great!